tests/utilities/test_data.py

```diff
@@ -23,6 +23,22 @@ def test_extract_batch_size():
     batch = {"test": [{"test": [torch.zeros(11, 10)]}]}
     assert extract_batch_size(batch) == 11
 
+    batch = {"test": [{"test": [torch.zeros(11, 10), torch.zeros(10, 10)]}]}
+    with pytest.warns(UserWarning, match="Trying to infer the `batch_size` .* we found is 11."):
+        extract_batch_size(batch)
+
+    warning_cache.clear()
+
+    batch = {"test": [{"test": [torch.zeros(10, 10), torch.zeros(11, 10)]}]}
+    with pytest.warns(UserWarning, match="Trying to infer the `batch_size` .* we found is 10."):
+        extract_batch_size(batch)
+
+    warning_cache.clear()
+
+    batch = [{"test": torch.zeros(10, 10), "test_1": torch.zeros(11, 10)}]
+    with pytest.warns(UserWarning, match="Trying to infer the `batch_size` .* we found is 10."):
```

Comment:
I think this test needs a check that no warning is raised if the collection has a uniform batch size